src/main/java/org/icatproject/ids/LockManager.java

```diff
@@ -55,6 +57,13 @@ void dec() {
 			count -= 1;
 			if (count == 0) {
 				lockMap.remove(id);
+				if (storageLock != null) {
```

Comment:
Why do we need to check `storageLock` to be `null` here?
Should this ever happen? 
If the lock fails an exception is thrown otherwise storageLock is a non-null value?

Reply:
Yes, this can happen. In fact, it will most of the time. If the storage plugin chooses not to implement file system locking at all, it may use a dummy implementation for `mainStorage.lock()` that simply always returns `null`. This is even the [default](https://github.com/icatproject/ids.plugin/blob/4b94d1076d40fd38a638d8b1f237083aeaa68429/src/main/java/org/icatproject/ids/plugin/AbstractMainStorage.java#L87) in class `AbstractMainStorage`.

Reply:
Thanks a lot.